src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -334,4 +377,67 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode>, pr
   // @}
 };
 
+/**
+ * LQP node types should derive from this, instead of directly from AbstractLQPNode, in order to enable the
+ * <NodeType>::make() that allows for a clean notation when building LQPs via code by allowing to pass in a nodes
+ * child(ren) as the last argument(s).
```

Comment:
Brrr. This puts some implementation detail into the class hierarchy. Why do we need this class?

Reply:
AbstractLQPNode needs to remain a non-template, so we can keep pointers to it, which is what we do everywhere.
Your proposal included `make` methods in every node type that forwarded the DerivedNodeType to make. Of course this is possible to, but I found a CRTP class to be way less text.
I think `AbstractLQPNodeEnableMake` doesn't even need to be derived from `AbstractLQPNode`, but without it you'd have a dozen node types inheriting from two classes, so I chose this path.

I agree it's not the most beautiful things, but it's not too bad either.

Reply:
In my opinion, the second option that you presented (multiple inheritance `public AbstractLQPNode, public EnableMakeForLQPNode`) is better because it keeps the class hierarchy clean. For me, that's enough of a reason.

Think `std::enable_shared_from_this`.

Reply:
Okay, will go for that solution then.